eclair-node/src/main/scala/fr/acinq/eclair/api/Service.scala

```diff
@@ -223,6 +222,7 @@ trait Service extends ExtraDirectives with Logging {
                         } ~
                         path("payinvoice") {
                           formFields(invoiceFormParam, amountMsatFormParam.?, "maxAttempts".as[Int].?, "feeThresholdSat".as[Satoshi].?, "maxFeePct".as[Double].?) {
+                            case (invoice, _, _, _, _) if invoice.isExpired => reject(MalformedFormFieldRejection("invoice", "The invoice provided has expired"))
```

Comment:
This isn't exactly a "malformed" field, but I'm not very familiar with akka http's DSL, do you have a suggestion on what to use there?

Reply:
I think `ValidationRejection` or even an `IllegalArgumentException` would be more appropriate, since the field is correctly formed but semantically invalid.